go-controller/pkg/node/node_smartnic.go

```diff
@@ -0,0 +1,203 @@
+func (n *OvnNode) getVfRepName(pod *kapi.Pod) (string, error) {
+	smartNicCD := util.SmartNICConnectionDetails{}
+	if err := smartNicCD.FromPodAnnotation(pod.Annotations); err != nil {
+		return "", fmt.Errorf("failed to get smart-nic annotation. %v", err)
+	}
+	return util.GetSriovnetOps().GetVfRepresentorSmartNIC(smartNicCD.PfId, smartNicCD.VfId)
+}
+
+// addRepPort adds the representor of the VF to the ovs bridge
+func (n *OvnNode) addRepPort(pod *kapi.Pod, vfRepName string, ifInfo *cni.PodInterfaceInfo) error {
+	klog.Infof("addRepPort: %s", vfRepName)
+	smartNicCD := util.SmartNICConnectionDetails{}
+	if err := smartNicCD.FromPodAnnotation(pod.Annotations); err != nil {
+		return fmt.Errorf("failed to get smart-nic annotation. %v", err)
+	}
+
+	err := cni.ConfigureOVS(context.TODO(), pod.Namespace, pod.Name, vfRepName, ifInfo, smartNicCD.SandboxId)
+	if err != nil {
+		return err
+	}
+	klog.Infof("Port %s added to bridge br-int", vfRepName)
+
+	link, err := util.GetNetLinkOps().LinkByName(vfRepName)
+	if err != nil {
+		// Note(adrianc): we are lenient with cleanup in this method as pod is going to be retried anyway.
+		_ = n.delRepPort(vfRepName)
+		return fmt.Errorf("failed to get link device for interface %s", vfRepName)
+	}
+
+	klog.Infof("addRepPort: set link mtu %s", vfRepName)
+	if err = util.GetNetLinkOps().LinkSetMTU(link, ifInfo.MTU); err != nil {
+		_ = n.delRepPort(vfRepName)
+		return fmt.Errorf("failed to setup representor port. failed to set MTU for interface %s", vfRepName)
+	}
+
+	klog.Infof("addRepPort: set link up for %s", vfRepName)
+	if err = util.GetNetLinkOps().LinkSetUp(link); err != nil {
+		_ = n.delRepPort(vfRepName)
+		return fmt.Errorf("failed to setup representor port. failed to set link up for interface %s", vfRepName)
+	}
+
+	// Update connection-status annotation
+	// TODO(adrianc): we should update Status in case of error as well
+	connStatus := util.SmartNICConnectionStatus{Status: util.SmartNicConnectionStatusReady, Reason: ""}
+	podAnnotator := kube.NewPodAnnotator(n.Kube, pod)
+	err = connStatus.SetPodAnnotation(podAnnotator)
+	if err != nil {
+		// we should not get here
+		_ = util.GetNetLinkOps().LinkSetDown(link)
+		_ = n.delRepPort(vfRepName)
+		return fmt.Errorf("failed to setup representor port. failed to set pod annotations. %v", err)
+	}
+
+	err = podAnnotator.Run()
+	if err != nil {
+		// cleanup
+		_ = util.GetNetLinkOps().LinkSetDown(link)
+		_ = n.delRepPort(vfRepName)
+		return fmt.Errorf("failed to setup representor port. failed to set pod annotations. %v", err)
+	}
+	return nil
+}
+
+// delRepPort delete the representor of the VF from the ovs bridge
+func (n *OvnNode) delRepPort(vfRepName string) error {
+	//TODO(adrianc): handle: clearPodBandwidth(pr.SandboxID), pr.deletePodConntrack()
+	klog.Infof("delRepPort: %s", vfRepName)
+	// Set link down for representor port
+	link, err := util.GetNetLinkOps().LinkByName(vfRepName)
```

Comment:
why do you need to set it down before you delete it?

Reply:
we can also do it after IMO. i will swap the order as first priority is to get it removed from OVS.